app/src/main/java/com/babylon/wallet/android/domain/usecases/GetSeedPhrasesWithAccountsUseCase.kt

```diff
@@ -0,0 +1,73 @@
+package com.babylon.wallet.android.domain.usecases
+
+import com.babylon.wallet.android.domain.model.DeviceFactorSourceData
+import kotlinx.collections.immutable.toPersistentList
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import rdx.works.profile.data.model.currentNetwork
+import rdx.works.profile.data.model.extensions.factorSourceId
+import rdx.works.profile.data.model.extensions.usesCurve25519
+import rdx.works.profile.data.model.extensions.usesSecp256k1
+import rdx.works.profile.data.model.factorsources.DeviceFactorSource
+import rdx.works.profile.domain.GetProfileUseCase
+import rdx.works.profile.domain.notHiddenAccounts
+import rdx.works.profile.domain.notHiddenPersonas
+import javax.inject.Inject
+
+class GetSeedPhrasesWithAccountsUseCase @Inject constructor(
```

Comment:
Maybe **GetFactorSourcesDataUseCase** ? The name implies that it returns seed phrases but it doesn't